Source/WebKit/UIProcess/Extensions/WebExtensionContext.h

```diff
@@ -542,6 +542,10 @@ class WebExtensionContext : public API::ObjectImpl<API::Object::Type::WebExtensi
     template<typename T>
     void sendToContentScriptProcessesForEvent(WebExtensionEventListenerType, const T& message) const;
 
+    void popupOpening() { s_hasOpenPopup = true; };
+    void popupClosed() { s_hasOpenPopup = false; };
+    bool popupIsOpen() const { return s_hasOpenPopup; };
```

Comment:
These would move to `WebExtensionController`, and can be two methods. `isShowingActionPopup()` and `setShowingActionPopup()` would be good names.